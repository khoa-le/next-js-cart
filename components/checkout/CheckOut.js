import { useReducer, useState } from "react";

import OrderSummary from "./OrderSummary";
import OrderForm from "./OrderForm";
import OrderSuccess from "./OrderSuccess";

const CheckOut = () => {
  const [showSuccessOrder, setShowSuccessOrder] = useState(false);
  const [successOrder, setSuccessOrder] = useState({});
  const initCheckoutState = { showSuccessOrder: false };

  const reducer = (initalState, action) => {
    if (action.type == "ORDER_SUCCESS") {
      setShowSuccessOrder(true);
      setSuccessOrder(action.data);
    }
  };

  const [checkoutState, dispatch] = useReducer(reducer, initCheckoutState);

  return (
    <div className="container">
      <div className="mb-9">
        <h1 className="h2 font-weight-normal">Thanh To√°n</h1>
      </div>
      <div className="row">
        {showSuccessOrder ? (
          <OrderSuccess order={successOrder} />
        ) : (
          <>
            <OrderSummary />
            <OrderForm checkoutDispath={dispatch} />
          </>
        )}
      </div>
    </div>
  );
};

export default CheckOut;
