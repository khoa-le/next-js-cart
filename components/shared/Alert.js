const Alert = ({ type, message }) => {
  const classAlert = `alert alert-${type}`;
  return (
    <div className={classAlert} role="alert">
      {message}
    </div>
  );
};

export default Alert;
