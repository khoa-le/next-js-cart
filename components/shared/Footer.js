export default () => (
  <footer>
    <div className="container">
      <hr className="my-0" />
      <div className="container text-center u-space-1">
        <p className="small text-muted">
          Â© 2020 Next Js Cart. All rights reserved.
        </p>
      </div>
    </div>
  </footer>
);
