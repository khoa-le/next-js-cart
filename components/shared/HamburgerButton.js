export default () => (
  <button
    className="navbar-toggler btn d-md-none u-hamburger"
    type="button"
    aria-label="Toggle navigation"
    aria-expanded="false"
    aria-controls="navBar"
    data-toggle="collapse"
    data-target="#navBar"
  >
    <span id="hamburgerTrigger" className="u-hamburger__box">
      <span className="u-hamburger__inner"></span>
    </span>
  </button>
);
