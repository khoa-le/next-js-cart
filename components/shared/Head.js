import Head from "next/head";

export default () => (
  <Head>
    <meta
      name="viewport"
      content="user-scalable=no,initial-scale=1,maximum-scale=1,minimum-scale=1,width=device-width,height=device-height"
    />
    <meta charSet="utf-8" />
    <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
    <link
      rel="icon" type="image/x-icon" 
      href="/assets/images/instacart_favicon-505fea7a2940344ebace028b21e929d5.ico"
    />
    <title>Next Js Cart</title>
  </Head>
);
