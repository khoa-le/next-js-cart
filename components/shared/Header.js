import TopBar from "./TopBar";
import LogoAndNav from "./LogoAndNav";

const Header = () => (
  <>
    <TopBar />
    <LogoAndNav/>
  </>
);
export default Header;
