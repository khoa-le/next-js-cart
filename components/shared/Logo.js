import Link from "next/link";

const Logo = () => (
    <Link href="/">
      <a className="navbar-brand u-header__navbar-brand" aria-label="Front">
        <img
          src="/assets/images/logo_transparent.png"
          width="150"
          alt="Next Js Cart"
        />
      </a>
    </Link>

);

export default Logo;
