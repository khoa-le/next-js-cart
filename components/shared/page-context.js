import React from "react";
export const PageContext = React.createContext({
  cart: { total: 0 },
  changeCart: () => {},
});
