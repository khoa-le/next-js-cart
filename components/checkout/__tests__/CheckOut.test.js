import { render, cleanup, act } from "@testing-library/react";
import CheckOut from "../CheckOut";
import { PageContext } from "../../shared/page-context";
jest.mock("../../../lib/hooks/useDataApi");

afterEach(cleanup);

const valueContext = {
  cart: {
    userCart: {
      cart_item: []
    }
  }
};

describe("CheckOut", () => {
  test("Can render CheckOut Form", async () => {
    await act(async () => {
      const { container, getByText } = render(
        <PageContext.Provider value={valueContext}>
          <CheckOut />
        </PageContext.Provider>
      );

      const headerElement = getByText("Thanh To√°n");
      expect(container.firstChild).toMatchSnapshot();
      expect(headerElement).toBeInTheDocument();
    });
  });
});
