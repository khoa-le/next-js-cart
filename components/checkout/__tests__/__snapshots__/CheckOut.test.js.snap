// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckOut Can render CheckOut Form 1`] = `
<div
  class="container"
>
  <div
    class="mb-9"
  >
    <h1
      class="h2 font-weight-normal"
    >
      Thanh Toán
    </h1>
  </div>
  <div
    class="row"
  >
    <div
      class="col-lg-4 order-lg-2 mb-9 mb-lg-0"
    >
      <div
        class="mb-4"
      >
        <h2
          class="h5"
        >
          Thông tin đơn hàng
        </h2>
      </div>
      <div
        class="shadow-sm rounded p-5"
      >
        <hr
          class="my-5"
        />
        <form
          class="js-focus-state input-group u-form"
        >
          <input
            aria-label="Discount"
            class="form-control u-form__input"
            placeholder="Discount"
            type="search"
          />
          <div
            class="input-group-append"
          >
            <button
              class="btn btn-primary u-btn-primary"
              type="button"
            >
              Apply
            </button>
          </div>
        </form>
        <hr
          class="my-5"
        />
        <div
          class="media align-items-center"
        >
          <div
            class="d-flex mr-3"
          >
            <h3
              class="h6 text-secondary"
            >
              Tạm Tính
            </h3>
          </div>
          <div
            class="media-body text-right"
          >
            <span />
          </div>
        </div>
        <div
          class="media align-items-center"
        >
          <div
            class="d-flex mr-3"
          >
            <h3
              class="h6 text-secondary"
            >
              Phí Vận Chuyển
            </h3>
          </div>
          <div
            class="media-body text-right"
          >
            <span>
              miễn phí
            </span>
          </div>
        </div>
        <hr
          class="my-5"
        />
        <div
          class="media align-items-center"
        >
          <div
            class="d-flex mr-3"
          >
            <h3
              class="h6 text-secondary"
            >
              Tổng Cộng
            </h3>
          </div>
          <div
            class="media-body text-right"
          >
            <strong />
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-lg-8 order-lg-1"
    >
      <form
        class="js-validate"
      >
        <div
          class="mb-4"
        >
          <h2
            class="h5"
          >
            Địa chỉ nhận hàng
          </h2>
        </div>
        <div
          class="row mb-9"
        >
          <div
            class="col-md-6"
          >
            <div
              class="js-form-message mb-6"
            >
              <label
                class="h6 small d-block text-uppercase"
              >
                Tên
                <span
                  class="text-danger"
                >
                  *
                </span>
              </label>
              <div
                class="js-focus-state input-group u-form"
              >
                <input
                  aria-label="Jack"
                  class="form-control u-form__input"
                  data-error-class="u-has-error"
                  data-msg="Please enter your frist name."
                  data-success-class="u-has-success"
                  placeholder="Jack"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="js-form-message mb-6"
            >
              <label
                class="h6 small d-block text-uppercase"
              >
                Họ
                <span
                  class="text-danger"
                >
                  *
                </span>
              </label>
              <div
                class="js-focus-state input-group u-form"
              >
                <input
                  aria-label="Wayley"
                  class="form-control u-form__input"
                  data-error-class="u-has-error"
                  data-msg="Please enter your last name."
                  data-success-class="u-has-success"
                  placeholder="Wayley"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="w-100"
          />
          <div
            class="col-md-6"
          >
            <div
              class="js-form-message mb-6"
            >
              <label
                class="h6 small d-block text-uppercase"
              >
                Email address
                <span
                  class="text-danger"
                >
                  *
                </span>
              </label>
              <div
                class="js-focus-state input-group u-form"
              >
                <input
                  aria-label="jackwayley@gmail.com"
                  class="form-control u-form__input"
                  data-error-class="u-has-error"
                  data-msg="Please enter a valid email address."
                  data-success-class="u-has-success"
                  placeholder="jackwayley@gmail.com"
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="js-form-message mb-6"
            >
              <label
                class="h6 small d-block text-uppercase"
              >
                Số điện thoại
                <span
                  class="text-danger"
                >
                  *
                </span>
              </label>
              <div
                class="js-focus-state input-group u-form"
              >
                <input
                  aria-label="+1 (062) 109-9222"
                  class="form-control u-form__input"
                  data-error-class="u-has-error"
                  data-msg="Please enter your last name."
                  data-success-class="u-has-success"
                  placeholder="+1 (062) 109-9222"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="w-100"
          />
          <div
            class="col-md-6"
          >
            <div
              class="js-form-message mb-6"
            >
              <label
                class="h6 small d-block text-uppercase"
              >
                Tỉnh/Thành Phố
                <span
                  class="text-danger"
                >
                  *
                </span>
              </label>
              <div
                class="js-focus-state input-group u-form"
              >
                <select
                  class="custom-select"
                >
                  <option
                    value=""
                  >
                    --Chọn Tỉnh/Thành Phố--
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="mb-6"
            >
              <label
                class="h6 small d-block text-uppercase"
              >
                Quận/Huyện
                <span
                  class="text-danger"
                >
                  *
                </span>
              </label>
              <select
                class="custom-select"
              >
                <option
                  value=""
                >
                  --Chọn Quận/Huyện--
                </option>
              </select>
            </div>
          </div>
          <div
            class="w-100"
          />
          <div
            class="col-md-12"
          >
            <div
              class="js-form-message mb-6"
            >
              <label
                class="h6 small d-block text-uppercase"
              >
                Địa chỉ
                <span
                  class="text-danger"
                >
                  *
                </span>
              </label>
              <div
                class="js-focus-state input-group u-form"
              >
                <input
                  aria-label="470 Lucy Forks"
                  class="form-control u-form__input"
                  data-error-class="u-has-error"
                  data-msg="Please enter a valid address."
                  data-success-class="u-has-success"
                  placeholder="470 Lucy Forks"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="mb-4"
        >
          <h2
            class="h5"
          >
            Hình thức thanh toán
          </h2>
        </div>
        <div
          class="btn-group btn-group-toggle mb-6"
        >
          <a
            class="js-animation-link btn btn-sm btn-outline-secondary u-btn-wide--sm active"
            href="#"
          >
            Thanh Toán Lúc Nhận Hàng
          </a>
          <a
            class="js-animation-link btn btn-sm btn-outline-secondary u-btn-wide--sm "
            href="#"
          >
            Chuyển Khoản
          </a>
        </div>
        <div
          class=""
          data-target-group="paymentMethods"
          id="COD"
          style="display: block;"
        />
        <div
          class="animated slideInUp"
          data-target-group="paymentMethods"
          id="BANK_TRANSFER"
          style="display: none;"
        >
          <div
            class="row"
          >
            <div
              class="col-md-8"
            >
              <div
                class="js-form-message mb-6"
              >
                <label
                  class="h6 small d-block text-uppercase"
                >
                  Thông tin chuyển khoản:
                </label>
                <div
                  class="js-focus-state input-group u-form"
                >
                  <p
                    class="h6 text-secondary"
                  >
                    <strong>
                      Ngân hàng Quốc Tế VIB
                    </strong>
                  </p>
                  <div
                    class="w-100"
                  />
                  <p
                    class="h6 text-secondary"
                  >
                    Chủ tài khoản: LÊ NGỌC KHOA
                  </p>
                  <div
                    class="w-100"
                  />
                  <p
                    class="h6 text-secondary"
                  >
                    Số tài khoản: 6000123456
                  </p>
                  <div
                    class="w-100"
                  />
                  <p
                    class="h6 text-secondary"
                  >
                    Chi nhánh: Quận 5
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="d-flex justify-content-between align-items-center"
        >
          <a
            style="cursor: pointer;"
          >
            <span
              class="fa fa-angle-left mr-2"
            />
             Quay Lại
          </a>
          <button
            class="btn btn-primary u-btn-primary transition-3d-hover"
            type="submit"
          >
            Đặt Hàng
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
